lua/tests/automated/utils_spec.lua

```diff
@@ -307,3 +307,92 @@ describe("transform_path", function()
     end, new_relpath "doc/mydoc.md", new_relpath "d/mydoc.md")
   end)
 end)
+
+describe("path_tail", function()
+  local function assert_tails(paths)
+    for _, path in ipairs(paths) do
+      it("gets the tail of " .. path, function()
+        local tail = vim.fn.fnamemodify(path, ":p:t")
+        eq(tail, utils.path_tail(path))
+      end)
+    end
+  end
+
+  if jit and jit.os:lower() == "windows" then
+    describe("handles windows paths", function()
+      local paths = {
+        [[C:\Users\username\AppData\Local\nvim-data\log]],
+        [[D:\Projects\project_folder\source_code.py]],
+        [[E:\Music\song.mp3]],
+        [[/home/usuario/documents/archivo.txt]],
+        [[/var/www/html/index.html]],
+        [[/mnt/backup/backup_file.tar.gz]],
+      }
+
+      assert_tails(paths)
+    end)
+  elseif jit and jit.os:lower() == "linux" then
+    describe("handles linux paths", function()
+      local paths = {
+        [[/home/usuario/documents/archivo.txt]],
+        [[/var/www/html/index.html]],
+        [[/mnt/backup/backup_file.tar.gz]],
+      }
+
+      assert_tails(paths)
+    end)
+  elseif jit and jit.os:lower() == "osx" then
+    describe("handles macos paths", function()
+      local paths = {
+        [[/Users/Usuario/Documents/archivo.txt]],
+        [[/Applications/App.app/Contents/MacOS/app_executable]],
+        [[/Volumes/ExternalDrive/Data/file.xlsx]],
+      }
+
+      assert_tails(paths)
+    end)
+  end
+end)
+
+describe("split_lines", function()
+  local function read_file(path)
+    local file = io.open(path, "rb")
+    if not file then
+      return nil
+    end
+    local content = file:read "*a"
+    file:close()
+    return content
+  end
+
+  local newline = [[./lua/tests/fixtures/split_lines/newline.txt]]
+  local carriage_newline = [[./lua/tests/fixtures/split_lines/carriage_newline.txt]]
```

Comment:
is it strictly necessary to use real files for these?
I think I would prefer just using lua strings if possible

Reply:
That is definitely better! Unnecessary testing fixtures were removed.